File: src/datamodel.rb
#!/usr/bin/env ruby

class Tablature

    attr_accessor :band;
    attr_accessor :song;
    attr_accessor :content;
    attr_accessor :multicol;

    # Create the object
    def initialize(band = "", song = "", content = "", multicol = "")
        @band = band
        @song = song
        @content = content
        @multicol = multicol
    end

    # returns tex code
    def toTex

        # header
        tex = "%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n";
        tex += "\\newpage\n";
        tex += "\\section{ #{band} - #{song} }\n";

        if multicol == true
            tex += "\\begin{multicols}{2}\n";
        end

        tex += "\\begin{verbatim}\n";
        # content
        tex += content;
        tex += "\n\\end{verbatim}\n";

        if multicol == true
            tex += "\\end{multicols}\n";
        end

        return tex;
    end

    # compares to another tablature
    def equalTo(otherTablature)
        if otherTablature.is_a?(Tablature)
            a = otherTablature.band.downcase + "" + otherTablature.song.downcase;
            b = @band.downcase + "" + @song.downcase;

            if a == b
                return true;
            end
        end
        return false;
    end

end

class Chord

    attr_accessor :name;
    attr_accessor :value;

    # Create the object
    def initialize(band = "", song = "")
        @name = name
        @value = value
    end

    # returns tex code
    def toTex

        tex = "%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n";
        # TODO

        return tex;
    end

    # compares to another chord
    def equalTo(otherChord)
        if otherChord.is_a?(Chord)
            a = otherChord.name.downcase + "" + otherChord.value.downcase;
            b = @name.downcase + "" + @value.downcase;
            if a == b
                return true;
            end
        end
        return false;
    end

end

###
### ############################################################################
###


if __FILE__ == $0

    tb1 = Tablature.new("Oasis","Wonderwall","This is the content",false);
    tb2 = Tablature.new("Oasis","Wonderwall","This is a repetition",true);
    tb3 = Tablature.new("Daft Punk","Me","This is the content",true);

    puts tb1.toTex
    puts "------------------------------------------------"
    puts tb3.toTex
    puts "------------------------------------------------"
    puts tb1.equalTo(tb2)
    puts tb1.equalTo(tb3)

end

File: python/src/datamodel.py


class Tablature:

    # Create the object
    def __init__(self, band = "", song = "", content = "", multicol = False):
        self.band = band
        self.song = song
        self.content = content
        self.multicol = multicol

    # returns tex code
    def toHtml(self):

        aux = ""
        aux += "<h1>" + self.band + " - " + self.song + "</h1>"
        aux += "<code>" + self.content.replace("","") + "</code>"
        return aux


    # compares to another tablature
    def equalTo(self,otherTablature):

        if isinstance(otherTablature, Tablature):
            a = otherTablature.band.lower() + "" + otherTablature.song.lower()
            b = self.band.lower() + "" + self.song.lower()
            if a == b:
                return True

        return False


class Chord:

    # Create the object
    def __init__(self, band = "", song = ""):
        self._name = name
        self._value = value

    # returns tex code
    def toTex(self):

        return "toTex method"
        '''
        tex = "%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%\n";
        # TODO

        return tex;
        '''

    # compares to another chord
    def equalTo(self,otherChord):

        if isinstance(otherChord, Chord):
            a = otherChord._name.lower() + "" + otherChord._value.lower()
            b = self._name.lower() + "" + self._value.lower()
            if a == b:
                return True

        return False


###
### ############################################################################
###

# test statements

tb1 = Tablature("Oasis","Wonderwall","This is the content",False)
tb2 = Tablature("Oasis","Wonderwall","This is a repetition",True)
tb3 = Tablature("Daft Punk","Me","This is the content",True)


print tb1.toHtml()
print "------------------------------------------------"
print tb3.toHtml()
print "------------------------------------------------"
print tb1.equalTo(tb2)
print tb1.equalTo(tb3)

